ndcube/mixins/sequence_plotting.py

```diff
@@ -0,0 +1,1198 @@
+import numpy as np
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import astropy.units as u
+from sunpy.visualization.imageanimator import ImageAnimatorWCS, LineAnimator
+
+from ndcube import utils
+
+__all__ = ['NDCubePlotMixin']
+
+NON_COMPATIBLE_UNIT_MESSAGE = \
+  "All sequence sub-cubes' unit attribute are not compatible with unit_y_axis set by user."
+
+class NDCubeSequencePlotMixin:
+    def plot(self, cubesequence, axes=None, plot_as_cube=False, plot_as_line=False,
+             plot_axis_indices=None, axes_coordinates=None, axes_units=None, data_unit=None,
+             **kwargs):
+        """
+        Visualizes data in the NDCubeSequence.
+
+        Based on the dimensionality of the sequence and value of image_axes kwarg,
+        a Line/Image Animation/Plot is produced.
+
+        Parameters
+        ----------
+        axes: `astropy.visualization.wcsaxes.core.WCSAxes` or ??? or None.
+            The axes to plot onto. If None the current axes will be used.
+
+        plot_as_cube: `bool`
+            If the sequence has a common axis, visualize the sequence as a single
+            cube where the sequence sub-cubes are sequential along the common axis.
+            This will result in the sequence being treated as a cube with N-1 dimensions
+            where N is the number of dimensions of the sequence, including the sequence
+            dimension.
+            Default=False
+
+        plot_axis_indices: `int` or iterable of one or two `int`.
+            If two axis indices are given, the sequence is visualized as an image or
+            2D animation, assuming the sequence has at least 2 dimensions.
+            (N.B. If plot_as_cube is True, the number of sequence dimensions is effectively
+            reduced by 1.)  The dimension indicated by the 0th index is displayed on the
+            x-axis while the dimension indicated by the 1st index is displayed on the y-axis.
+            If only one axis index is given (either as an int or a list of one int),
+            then a 1D line animation is produced with the indicated dimension on the x-axis
+            and other dimensions represented by animations sliders.
+            Default=[-1, -2].  If sequence only has one dimension (or effectively one if
+            plot_as_cube is True), plot_axis_indices is ignored and a staice 1D line plot
+            is produced.
+
+        axes_coordinates: `list` of physical coordinates for image axes and sliders or `None`
+            If None coordinates derived from the WCS objects will be used for all axes.
+            If a list, it should contain one element for each axis.  Each element should
+            be either an `astropy.units.Quantity` or a `numpy.ndarray` of coordinates for
+            each pixel, or a `str` denoting a valid extra coordinate.
```

Comment:
Error in this kwarg description.  Elements can also be None.

axes_coordinates should probably also accept length 2 for images and length one for line plots so slider units don't have to be changed every time for high dimension sequences.